source/appModules/explorer.py

```diff
@@ -299,3 +300,16 @@ def event_gainFocus(self, obj, nextHandler):
 			return
 
 		nextHandler()
+
+	def isGoodUIAWindow(self,hwnd):
+		# #9204: In build 18305 and later, it is the shell that raises window open event for emoji panel.
+		if winVersion.winVersion.build > 17763 and winUser.getClassName(hwnd) == "ApplicationFrameWindow":
+			return True
+		return False
+
+	def event_UIA_window_windowOpen(self, obj, nextHandler):
+		# Send UIA window open event to input app window.
+		inputPanelWindow = obj.firstChild
+		if isinstance(obj, UIA) and obj.UIAElement.cachedClassName == "ApplicationFrameWindow" and inputPanelWindow and inputPanelWindow.appModule.appName == "windowsinternal_composableshell_experiences_textinput_inputapp":

```

Comment:
Could you split this into multiple lines? Also, I'd prefer checking the validity of obj before (and not after) getting the first child of it and checking prerequisites on that.